File: clientes-app/src/app/models/cliente.ts
export interface Cliente {
  id: number;
  nombre: string;
  apellido: string;
  telefono: string;
  correo: string;
  fechaNacimiento: string;
  activo: boolean;
  fechaCreacion: string;
}

export interface ClienteInput {
  nombre: string;
  apellido: string;
  telefono: string;
  correo: string;
  fechaNacimiento: string;
  activo: boolean;
}

export interface ResultadoPaginado<T> {
  datos: T[];
  pagina: number;
  tamanoPagina: number;
  totalRegistros: number;
  totalPaginas: number;
}

File: ClintesApi/ClintesApi/Models/Cliente.cs
namespace ClientesApi.Models;

public class Cliente
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Apellido { get; set; } = string.Empty;
    public string Telefono { get; set; } = string.Empty;
    public string Correo { get; set; } = string.Empty;
    public DateTime FechaNacimiento { get; set; }
    public bool Activo { get; set; } = true;            // Estado: Activo/Inactivo
    public DateTime FechaCreacion { get; set; } = DateTime.Now;
}